krabs/krabs/ut.hpp

```diff
@@ -113,19 +122,19 @@ namespace krabs { namespace details {
         // TODO: Only forward the calls that are requested to each provider.
         for (auto &provider : trace.providers_) {
             if (provider_flags.find(provider.get().guid_) != provider_flags.end()) {
-                provider_flags[provider.get().guid_].flags_tuple_ = std::make_tuple<UCHAR, ULONGLONG, ULONGLONG, UCHAR> (0, 0, 0, 0);
+                provider_flags[provider.get().guid_].filter_flags_ = {};
             }
 
-            std::get<0>(provider_flags[provider.get().guid_].flags_tuple_) |= provider.get().level_;
-            std::get<1>(provider_flags[provider.get().guid_].flags_tuple_) |= provider.get().any_;
-            std::get<2>(provider_flags[provider.get().guid_].flags_tuple_) |= provider.get().all_;
-            std::get<3>(provider_flags[provider.get().guid_].flags_tuple_) |= provider.get().trace_flags_;
+            provider_flags[provider.get().guid_].filter_flags_.level_ |= provider.get().level_;
```

Comment:
@kallanreed you had a comment here - 

```Do the lookup once and store it in a reference variable. That way you don't need to do the lookup in the map multiple times. Also it would let you get rid of the for loop below (for event_id filtering). If you end up using the std::set instead of the vector, then you shouldn't need all that looping anyway (use std::copy with an inserter)```

...could you elaborate? I didn't quite understand the suggestion.

Thanks!

Reply:
Something like this
```
auto& flags = provider_flags[provider.get().guid_];
std::get<1>(flags.flags_tuple_) |= provider.get().any_;
...
```
Even better if you get rid of that `std::variant`